torch/_refs/__init__.py

```diff
@@ -680,6 +682,10 @@ def tan(a):
 def tanh(a):
     return prims.tanh(a)
 
+@_make_elementwise_unary_reference(ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+                                   aten_op=None)
+def bfloat16(a):
+    return prims.convert_element_type(a, torch.bfloat16)
```

Comment:
A couple interesting issues here:

- the type promotion specified in `make_elementwise_unary_reference` is not completely correct, because this also casts complex tensor to float. I think that decorator can be dropped?
- type annotations are needed (`a` is a `Tensor` and the function returns a `Tensor`)
- when `a` is already in the bfloat16 dtype it's just returned without creating a copy, but I think `convert_element_type` always creates a copy? If so, this needs a short-circuit for the case that `a` is already bfloat16